#!/bin/bash -l
#SBATCH -A uppmax2025-3-3
#SBATCH -M snowy
#SBATCH -p core
#SBATCH -n 8
#SBATCH --mem=64G
#SBATCH -t 06:00:00
#SBATCH -J yahs-contactmap
#SBATCH --mail-type=ALL
#SBATCH --mail-user=mama8042@student.uu.se
#SBATCH --output=%x.%j.out

# Load required modules
module load bioinfo-tools

# Define paths
YAHSDIR="/proj/uppmax2025-3-3/Genome_Analysis/yahs"
ASSEMBLY="/home/isamaru/genome-analysis/DNA_analysis/01_DNA_assembly/polished_assembly.fasta"
FAI="${ASSEMBLY}.fai"
SCA="/home/isamaru/genome-analysis/DNA_analysis/04_DNA_scaffolding"
BIN="${SCA}/scaffolds.bin" # Already generated by yahs
AGP="${SCA}/scaffolds_scaffolds_final.agp"
OUTDIR="${SCA}/heatmaps"
JUICER_TOOLS="${YAHSDIR}/juicer_tools_1.22.01.jar"

mkdir -p $OUTDIR
cd $OUTDIR

# STEP 1: Generate sorted alignment file for juicer_tools

#${YAHSDIR}/juicer pre $BIN $AGP $FAI | \
   # sort -k2,2d -k6,6d -T ./ --parallel=8 -S32G | \
  #  awk 'NF' > alignments_sorted.txt.part && \
 #   mv alignments_sorted.txt.part alignments_sorted.txt

# STEP 2: Create scaffold size file (required by juicer_tools)

#cut -f1,2 $SCA/scaffolds_scaffolds_final.fa > scaffolds_final.chrom.#sizes

# STEP 3: Generate .hic file
java -Xmx32G -jar $JUICER_TOOLS pre \
    alignments_sorted.txt out.hic.part scaffolds_final.chrom.sizes && \
    mv out.hic.part yahs_output.hic
